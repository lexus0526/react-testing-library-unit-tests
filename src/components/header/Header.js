import React from "react";

export default function Header({test}) {
    return (
        <header>
            <h2 data-testid="h2tag" className="fancy-h1">{test}</h2>
        </header>
    );
}
