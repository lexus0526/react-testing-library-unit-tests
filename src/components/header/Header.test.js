import React from 'react'
import ReactDOM from 'react-dom'
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Header from "./Header";

afterEach(cleanup)

it("renders", () => {
    const {asFragment} = render(<Header text="Hello!"/>)
    expect(asFragment()).toMatchSnapshot();
})

it("inserts text in h1", () => {
    const { getByTestId, getByText } = render(<Header text="Zhenja!" />)
    //expect(getByTestId("h2tag")).toHaveTextContent("Zhenja!")
    //expect(getByText('Zhenja!')).toHaveClass("fancy-h1")
    expect(getByTestId('h2tag')).toHaveClass("fancy-h1")
})
