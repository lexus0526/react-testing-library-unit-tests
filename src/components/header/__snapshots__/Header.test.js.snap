// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h2
      class="fancy-h1"
      data-testid="h2tag"
    />
  </header>
</DocumentFragment>
`;
